domain/src/coaching_session.rs

```diff
@@ -1,28 +1,117 @@
+use super::error::{DomainErrorKind, Error, ExternalErrorKind, InternalErrorKind};
 use entity::coaching_sessions::Model;
-use entity_api::{coaching_session, error::Error};
+use entity_api::{coaching_relationship, coaching_session, organization};
+use log::*;
 use sea_orm::DatabaseConnection;
+use serde_json::json;
+use std::env;
 
 pub async fn create(
     db: &DatabaseConnection,
     coaching_session_model: Model,
 ) -> Result<Model, Error> {
-    coaching_session::create(db, coaching_session_model).await
+    let coaching_relationship =
+        coaching_relationship::find_by_id(db, coaching_session_model.coaching_relationship_id)
+            .await?;
+    let organization = organization::find_by_id(db, coaching_relationship.organization_id).await?;
+    let document_name = format!(
+        "{}.{}.{}",
+        organization.slug,
+        coaching_relationship.slug,
+        coaching_session_model.date.and_utc().timestamp()
+    );
+    info!(
+        "Attempting to create Tiptap document with name: {}",
+        document_name
+    );
+    let tip_tap_url = env::var("TIP_TAP_URL").map_err(|err| {
+        warn!("Failed to get Tiptap URL from environment: {:?}", err);
+        Error {
+            source: Some(Box::new(err)),
+            error_kind: DomainErrorKind::Internal(InternalErrorKind::Other),
+        }
+    })?;
+    let full_url = format!(
+        "{}/api/documents/{}?format=json",
+        tip_tap_url, document_name
+    );
+    let client = tip_tap_client().await?;
+
+    let request = client
+        .post(full_url)
+        .json(&json!({"type": "doc", "content": []}));
+    let response = match request.send().await {
+        Ok(response) => {
+            info!("Tiptap response: {:?}", response);
+            response
+        }
+        Err(e) => {
+            warn!("Failed to send request: {:?}", e);
+            return Err(e.into());
+        }
+    };
+
+    // Tiptap's API will return a 200 for successful creation of a new document
+    // and will return a 409 if the document already exists. We consider both "successful".
+    if response.status().is_success() || response.status().as_u16() == 409 {
```

Comment:
Likely will think through this a bit more as an existing TipTap doc likely means we attempted to create a coaching session as previously as well. For now I think this works.